packages/ember-htmlbars/lib/keywords/each.js

```diff
@@ -3,18 +3,10 @@
 @submodule ember-htmlbars
 */
 
-import ArrayController from 'ember-runtime/controllers/array_controller';
-
 export default function each(morph, env, scope, params, hash, template, inverse, visitor) {
   let getValue = env.hooks.getValue;
-  let firstParam = params[0] && getValue(params[0]);
   let keyword = hash['-legacy-keyword'] && getValue(hash['-legacy-keyword']);
 
-  if (firstParam && firstParam instanceof ArrayController) {
```

Comment:
I believe these hooks will need to stay (as deprecated) for some reasonable period of 2.x.

Obviously requiring the `ArrayController` for the `instanceof` wont work, so we will need to revert to duckTyping of some kind. Maybe something like: `firstParam.isArrayController`  ?